met-api/src/met_api/services/engagement_service.py

```diff
@@ -31,15 +31,23 @@ class EngagementService:
     otherdateformat = '%Y-%m-%d'
 
     @staticmethod
-    def get_engagement(engagement_id, user_id) -> EngagementSchema:
+    def get_engagement(engagement_id) -> EngagementSchema:
         """Get Engagement by the id."""
         engagement_model: EngagementModel = EngagementModel.find_by_id(engagement_id)
 
         if engagement_model:
-            if user_id is None \
+            if TokenInfo.get_id() is None \
                     and engagement_model.status_id not in (Status.Published.value, Status.Closed.value):
```

Comment:
Do you think (Status.Published.value, Status.Closed.value) should be passed as argument. This condition might change I think. (Status.Published.value, Status.Closed.value)  could be the default maybe

Reply:
i think i can make it to a separate method and improve how it looks like i think..
If the calling method has to pass the status , it might be like the calling method decides the security..so someone might call the method not knowing whats the status and override... thats the challenege